cli/command_server_start.go

```diff
@@ -273,6 +277,24 @@ func (c *commandServerStart) run(ctx context.Context) error {
 	return errors.Wrap(srv.SetRepository(ctx, nil), "error setting active repository")
 }
 
+// shutdownServer shutdown http server and close the repository.
+func shutdownServer(ctx context.Context, httpServer *http.Server, srv *server.Server) {
+	log(ctx).Infof("Shutting down...")
+
+	if serr := httpServer.Shutdown(ctx); serr != nil {
+		log(ctx).Debugf("unable to shut down http server: %v", serr)
+	}
+
+	rep := srv.GetRepository()
+	if rep == nil {
+		return
+	}
+
+	if rerr := rep.Close(ctx); rerr != nil {
```

Comment:
I am not sure whether Close() would be balanced with shutdownServer() teardown path. How important is it to close the repo during teardown ? What happens if we do not call close() ?

IMO a Close() should only be performed by someone who either does an addRef() o the repo object or who creates the repo object. Since, shutdownServer() did not open() it, so it should not be calling close() on it either. What do you think about this style of maintaining the resource construction and destruction layering, Aaron ?